addons/sale_timesheet/models/sale_order.py

```diff
@@ -94,22 +112,37 @@ def _get_prepaid_service_lines_to_upsell(self):
 
     def action_view_timesheet(self):
         self.ensure_one()
+        if not self.order_line:
+            return {'type': 'ir.actions.act_window_close'}
+
         action = self.env["ir.actions.actions"]._for_xml_id("sale_timesheet.timesheet_action_from_sales_order")
+        default_sale_line = next(filter(lambda sol: sol.is_service and sol.product_id.service_policy in ['ordered_prepaid', 'delivered_timesheet'], self.order_line))
```

Comment:
For consistency's sake, I would use a generator expression here rather than a filter ⁽ˢᵒʳʳʸ⁾